### PR TITLE
Make the crate `no_std`

### PR DESCRIPTION
**Pull request by lgiussan**

### CODE REVIEW

**Review by DanieleDiBenedetto — APPROVED** [review]

File: src/bin/generate-sample-proof.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
+// Copyright 2024, Horizen Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
```
Comment (la10736) [inline src/bin/generate-sample-proof.rs:6]: Run this script generate `proof.bin`, `pubs.bin` and `vk.bin` in the root: can you add them to `.gitignore`?

File: src/serde.rs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+        // Deserialize MyMap from an abstract "map" provided by the
+        // Deserializer. The MapAccess input is a callback provided by
+        // the Deserializer to let us see each entry in the map.
+        fn visit_map<M>(self, mut access: M) -> Result<Self::Value, M::Error>
+        where
+            M: MapAccess<'de>,
+        {
+            let mut map =
+                IndexMap::with_capacity_and_hasher(access.size_hint().unwrap_or(0), <_>::default());
```
Comment (la10736) [inline src/serde.rs:132]: Can you use `Default::default()` instead?

File: src/serde.rs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    table: OwnedTable<DoryScalar>,
+    verification_hash: [u8; 32],
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(remote = "OwnedTable<DoryScalar>")]
+pub struct OwnedTableDef {
+    #[serde(getter = "OwnedTable::inner_table", with = "index_map_serde")]
+    table: IndexMap,
+}
```
Comment (la10736) [inline src/serde.rs:50]: Sure? You really need it? Seams that `indexmap` crate already have a `serde` feature. Did you have tried it?

**Review by la10736 — CHANGES_REQUESTED** [review]
More related to ci and build process. Beside that replace `serde_cbor` with some maintained package and check if yu can use the `indexmap` built-in serde support.

**Review by la10736 — APPROVED** [review]
GREAT JOB

**Pull request merged**